### PR TITLE
Cleanup nosekatreport before migrating to Python 3

### PR DESCRIPTION
**Pull request by mmphego**
This PR does the following:
- [x] Adds BSD license 
- [x] Fix broken and remove unused tests

### CODE REVIEW

File: nosekatreport/test_template.py [hunk 1]
```diff
@@ -1,69 +0,0 @@
-###############################################################################
```
Comment (mmphego) [inline nosekatreport/test_template.py:1]: @ajoubertza what's your take on the files deleted?
Most of the functions used in this file aren't implemented ie `tests`, `report_subpass`, `report_subfail` and `report_progress`.

File: test_working/test_module2.py [hunk 2]
```diff
@@ -1,53 +0,0 @@
-###############################################################################
```
Comment (mmphego) [inline test_working/test_module2.py:1]: Ditto

File: nosekatreport/test_template.py (hunk 1 above)
Comment (ajoubertza) [reply nosekatreport/test_template.py:1]: The report methods look a bit like `Aqf.hop`, `Aqf.fail`, and `Aqf.step`.  You could use those instead.  If these tests weren't being run, then could be removed.

File: LICENSE [hunk 3]
```diff
@@ -1,21 +1,32 @@
-MIT License
+GNU GENERAL PUBLIC LICENSE
```
Comment (ajoubertza) [inline LICENSE:1]: The original request to open source this package recommended BSD-3 clause, so that would be better.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,89 @@
+pipeline {
```
Comment (ajoubertza) [inline Jenkinsfile:1]: Why do we need Jenkins test and Travis tests?

File: nosekatreport/__init__.py [hunk 5]
```diff
@@ -1,2 +1,16 @@
+"""Root of the nosekatreport package."""
+
+# BEGIN VERSION CHECK
+# Get package version when locally imported from repo or via -e develop install
+try:
+    import katversion as _katversion
```
Comment (ajoubertza) [inline nosekatreport/__init__.py:6]: Good idea to use katversion.

File: setup.py [hunk 6]
```diff
@@ -6,39 +6,46 @@
...
-    license='GPL',
-    packages=find_packages(exclude=['ez_setup']),
-    install_requires=['Nose>=0.11.0', 'traceback2', 'ansicolors', 'numpy'],
-    url='',
+    name="nosekatreport",
+    description="Nose plugin for writing an annotated test report.",
+    long_description=open("README.rst").read(),
+    author="CAM team",
+    author_email="cam@ska.ac.za",
+    license="GPL",
```
Comment (ajoubertza) [inline setup.py:18]: BSD

File: nosekatreport/test_template.py (hunk 1 above)
Comment (mmphego) [reply nosekatreport/test_template.py:1]: Tests weren't being run, also `katscripts` doesn't use any of those methods/functions.

**Review by ajoubertza — CHANGES_REQUESTED** [review]
Would prefer to split the cleanup work and Python 2 to 3 work into 2 PRs.  It looks like that is your plan.  Then PR title needs to change, and tests should only be targeting 2.7 for now.

**Comment by mmphego** [comment]
Closing PR

**Pull request closed without merging**